pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -0,0 +1,504 @@
+
+    _hdfsDefaultClasspathDir = _properties.getProperty("hdfs.default.classpath.dir", null);
+    _defaultPermissionsMask = _properties.getProperty(JobConfigConstants.DEFAULT_PERMISSIONS_MASK, null);
+
+    // Optional push location and table parameters. If set, will use the table config and schema from the push hosts.
+    String pushHostsString = _properties.getProperty(JobConfigConstants.PUSH_TO_HOSTS);
+    String pushPortString = _properties.getProperty(JobConfigConstants.PUSH_TO_PORT);
+    if (pushHostsString != null && pushPortString != null) {
+      _pushLocations =
+          PushLocation.getPushLocations(StringUtils.split(pushHostsString, ','), Integer.parseInt(pushPortString));
+    } else {
+      _pushLocations = null;
+    }
+
+    _logger.info("*********************************************************************");
+    _logger.info("enable.partitioning: {}", _enablePartitioning);
+    _logger.info("enable.sorting: {}", _enableSorting);
+    _logger.info("enable.resizing: {}", _enableResizing);
+    _logger.info("path.to.input: {}", _inputSegmentDir);
+    _logger.info("path.to.output: {}", _preprocessedOutputDir);
+    _logger.info("hdfs.default.classpath.dir: {}", _hdfsDefaultClasspathDir);
+    _logger.info("push.locations: {}", _pushLocations);
+    _logger.info("*********************************************************************");
+  }
+
+  public void run() throws Exception {
+    if (!_enablePartitioning && !_enableSorting && !_enableResizing) {
+      _logger.info("Pre-computing job is disabled.");
+      return;
+    } else {
+      _logger.info("Starting {}", getClass().getSimpleName());
+    }
+
+    _fileSystem = FileSystem.get(_configuration);
+    final List<String> inputDataPath = Arrays.asList(_inputSegmentDir.split("\\s*,\\s*"));
+    Path outputDir = new Path(_preprocessedOutputDir);
+
+    if (_fileSystem.exists(outputDir)) {
+      _logger.warn("Found the output folder {}, deleting it", _preprocessedOutputDir);
+      _fileSystem.delete(outputDir, true);
+    }
+    setDirPermission(outputDir);
+
+    _tableConfig = getTableConfig();
+
+    if (_enablePartitioning) {
+      fetchPartitioningConfig();
+      _logger.info("partition column: {}", _partitionColumn);
+      _logger.info("number of partitions: {}", _numberOfPartitions);
+      _logger.info("partition column: {}", _partitionColumn);
+    }
+
+    if (_enableSorting) {
+      fetchSortingConfig();
+      _logger.info("sorted.column: {}", _sortedColumn);
+    }
+
+    if (_enableResizing) {
+      fetchResizingConfig();
+      _logger.info("minimum number of output files: {}", _numberOfOutputFiles);
+    }
+
+    _logger.info("Initializing a pre-computing job");
+    Job job = Job.getInstance(_configuration);
+
+    job.getConfiguration().set(JobContext.JOB_NAME, this.getClass().getName());
+    job.getConfiguration().set(MRJobConfig.MAPREDUCE_JOB_USER_CLASSPATH_FIRST, "true");
```

Comment:
please leave a comment about this param, (also, do we still need it with your latest avro fix?)

Reply:
It's best to always turn on this setting, since there could be other dependency jars that is required higher versions.

Reply:
ok, may be good to document so users know that

Reply:
Done.